src/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -14,11 +14,9 @@ internal static class ConsolePal
     {
         private const int DefaultConsoleBufferSize = 256; // default size of buffer used in stream readers/writers
 
-        private static IntPtr s_InvalidHandleValue = new IntPtr(-1);
+        private static IntPtr InvalidHandleValue => new IntPtr(-1);
 
-        private static bool s_isWindows7 = GetIsWindows7();
```

Comment:
I have noticed initialization of `s_isWindows7` running during a simple `Console.WriteLine("Hello world")`. The OS version is cached by Environment already, and IsWindows7 check is on rare path in Title property. It is fine better to just recompute it each time it is needed without caching.